src/citrine/resources/dataset.py

```diff
@@ -289,28 +289,33 @@ def delete(self, data_concepts_resource: ResourceType, dry_run=False) -> Resourc
         return self._collection_for(data_concepts_resource) \
             .delete(uid[1], scope=uid[0], dry_run=dry_run)
 
-    def gemd_batch_delete(self, id_list: List[Union[LinkByUID, UUID]]) -> \
+    def gemd_batch_delete(self, id_list: List[Union[LinkByUID, UUID, str, BaseEntity]]) -> \
             List[Tuple[LinkByUID, ApiError]]:
         """
         Remove a set of GEMD objects.
 
         You may provide GEMD objects that reference each other, and the objects
-        will be removed in the appripriate order.
+        will be removed in the appropriate order.
 
         A failure will be returned if the object cannot be deleted due to an external
         reference.
 
         All data objects must be associated with this dataset resource. You must also
         have write access on this dataset.
 
+        If you wish to delete more than 50 objects, queuing of deletes requires that
+        the types of objects be known, and thus you _must_ provide ids in the form
+        of BaseEntities.
```

Comment:
Long-term I suppose we could also query that in a batch-get first? For now this seems expedient.